starlette/applications.py

```diff
@@ -74,7 +74,7 @@ def __init__(
             {} if exception_handlers is None else dict(exception_handlers)
         )
         self.user_middleware = [] if middleware is None else list(middleware)
-        self.middleware_stack = self.build_middleware_stack()
+        self.middleware_stack: typing.Optional[ASGIApp] = None
```

Comment:
IMO, `build_middleware_stack` is a very cheap function to call. It makes sense to call it here and call once again when you need it again, and avoid nullable attributes (and null-checks).

Reply:
Moreover, it is not expected to be called often at runtime and affects only startup (configuration) time. 
So calling it multiple times seems ok to me.